### DISCUSSION TITLE
For Anyone Having Trouble Building!

### DISCUSSION THREAD

**Discussion by ivanogames** [qnblackcat/uYouPlus#1383] [General]
The OG guide is a little outdated and doesn't work for a lot of people, so in the interest of helping others as I have with the previous, I'll post an update I hope will work for most folks. For a TL;DR version, [follow the instructions from YTLitePlus github](https://github.com/YTLitePlus/YTLitePlus/wiki/Building). 

It should also be noted that downloading an [IPA directly from YTLitePlus](https://ytliteplus.github.io/) works. How they're able to keep it up, I dunno. Anyway, here goes if you insist on building:

**Instructions:**
1) Log in to GitHub or create an account and then sign in.
2) Search uYouPlus on Google > click "qnblackcat/uYouPlus - GitHub" > click "Fork" on the top right > Click the green "Create Fork" button. On the top left, you should now see "your user name / uYouPlus".
3) Go to the "Actions" tab near the top of the page. Click the green "I understand my workflows..." blah blah. 
4) Click "Build and Release uYouPlus" on the left-hand side. Let this webpage sit as is and open a new tab.

**NEW**

5) On Google, search "decrypted iOS IPA store," and click the link that has "armconverter" in the URL. Search for YouTube. In order to download the decrypted IPA, you'll have to have an account with that site, so create one and download the latest YouTube IPA.
6) Upload the decrypted IPA to [this litterbox cloud site](https://litterbox.catbox.moe/). Once it's done uploading, click the link it just created to copy the URL for the decrypted IPA cloud link.
7) Go back to the GitHub page and click "Run workflow" on the right-hand side.
8) Paste the link from the litterbox site. Do not modify anything on the link. 
12) Run workflow and let sit until a green check appears.
13) Click the "Build and Release..." link right next to the green check. Your IPA should be there somewhere.

**---EDIT 09/02/2024---UPDATED GUIDE ABOVE---**

This first part is for the folks who run the GitHub page for uYou and any other variations of it. I've got some suggestions. Freeloaders (like myself) skip down to Instructions. I want to preface this by saying that I got the decrypted IPA from the repository available through qnblackcat/Dat Nguyen, and that same IPA ended up working, so I know I did that part right. Google Drive links do not seem to work, however; I tried every imaginable link Google Drive gives you, and none of them worked. Switching to Dropbox and getting the download link from that site worked for me. We need more detailed instructions, so here's my little contribution.

**Instructions:**
1) Log in to GitHub or create an account and then sign in.
2) Search uYouPlus on Google > click "qnblackcat/uYouPlus - GitHub" > click "Fork" on the top right > Click the green "Create Fork" button. On the top left, you should now see "your user name / uYouPlus".
3) Go to the "Actions" tab near the top of the page. Click the green "I understand my workflows..." blah blah. 
4) Click "Build and Release uYouPlus" on the left-hand side. Let this webpage sit as is and open a new tab.
5) Search qnblackcat on Google > Click the result that says "Nguyen Dat qnblackcat" > Click the "Decrypted-IPAs" repository > OneDrive > select the "YouTube" file.
6) Select your desired YouTube IPA. For the more recent ones, make sure to sort by "Newer to Older" under the "Modified" tab.
7) Download the IPA to your desktop or a place you'll remember.
8) Log in to Dropbox (Yes, it has to be Dropbox) or create a Dropbox account if you do not have one. Once again, yes, it has to be Dropbox.
8a) If you created a Dropbox account, be sure to verify your email shortly after creating it. This step is necessary!
9) Upload the decrypted IPA to Dropbox. On the icon for the IPA, click "Share" > make sure "Anyone with this link can view" shows up > click the blue "Copy Link" button.
10) Go back to the GitHub page and click "Run workflow" on the right-hand side.
11) Paste the link from Dropbox into the second box. **At the end of the link, CHANGE** "dl=0" to "dl=1". **That is, just delete the 0 and type 1.**
12) Run workflow and let sit until a green check appears.
13) Click the "uYouPlus" text on the top left > Click the bold text that says "Releases" on the right-hand side of the page. Your IPA should be there.

Edit: Messed up my numbers lol.

**Comment by Hackr1301** [comment]
for step number 5 I cant find the link

**Reply by Houseoflight** [reply]
here.
https://github.com/qnblackcat
the decrypted ipa file is the third one at the bottom.

**Comment by Houseoflight** [comment]
> Paste the link from Dropbox into the second box. At the end of the link, CHANGE "dl=0" to "dl=1". That is, just delete the 0 and type 1.
heads up, when i did this it was stuck on an endless 5+ hour loop of "waiting for runner to pick up this job"... and when I reverted it back to 0 it actually ran like normal. 
no clue why but ya.

edit: the artifacted ipa didn't work properly but now for some reason changing "dl=0" ro "dl=1" runs successfully? so im not sure why the loop happened lol

**Comment by SharonVarughese** [comment]
Thank you so muchhh! Ur a lifesaver!!!

**Comment by amjunior20** [comment]
I had given up trying, but I managed thanks to the post

**Comment by ghost** [comment]
Build always fails even when using google drive or dropbox.

End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of YouTube.zip or
        YouTube.zip.zip, and cannot find YouTube.zip.ZIP, period.

**Reply by ivanogames** [reply]
Guide updated. I hope it helps!

**Comment by 54437698** [comment]
Same here, fails with error code 9

**Reply by ivanogames** [reply]
Guide updated. I hope it helps!

**Comment by opensourcefan2** [comment]
Process completed with exit code 2. How to solve this?

**Reply by ivanogames** [reply]
Guide updated. I hope it helps!

**Comment by CamelatBlue** [comment]
no green check. red x and it says "Process completed with exit code 2"

**Reply by ivanogames** [reply]
Guide updated. I hope it helps!

**Comment by frezzyQ** [comment]
THANK YOU! everything worked perfectly

**Comment by RainsDFS** [comment]
Thank you so much finally got this working!

**Comment by Cupo00** [comment]
Hi all, Ive seen in the decrypted ipas that he has a spotify one? Is there a no ads spotify app for sideloading aswell? Ive seen a couple on google but idk which ones are good.

**Comment by masont1017** [comment]
i cant open youtube 🤦‍♂️

**Comment by davidrittri** [comment]
Followed all the steps and it seemed to work, but I can't open youtube now, it crashes every time.

**Reply by ivanogames** [reply]
That seems to be an issue with the app itself, honestly. Not sure there's much we can do besides switch to uYou+ Extra, uYouEnhanced, YTLitePlus, etc. until the bugs are worked out. Anyway, guide updated. I hope it helps!

**Comment by framts** [comment]
same, i did everything like the building wiki said but now youtube won't open, it just keeps crashing

**Reply by MuhammadAliKhan1** [reply]
did you find a workaround or fixed it ?

**Comment by MuhammadAliKhan1** [comment]
Youtube keeps crashing

**Comment by komzuald** [comment]
hi, thx for answer. Already everything works fine :)Wiadomość napisana przez AtTimesChris ***@***.***> w dniu 11 sie 2024, o godz. 00:58:﻿
Hi i downloaded the most up to date youtube file and finished with a green tick but it isnt showing on releases under my fork?

**Comment by EliteArrow** [comment]
It worked, Thank you very much. You are the best broh.. ❤️🤍

**Comment by Stidgy** [comment]
This should be in the Wiki-Building not in discussion...Good job man

**Reply by ivanogames** [reply]
Good point. I missed that section of threads

**Comment by SnowfallOCEC** [comment]
I feel like im going crazy. I've followed your updated guide to the letter and keep running into this error here
![image](https://github.com/user-attachments/assets/30401e73-b179-4d7b-8ef8-5be5d61eeb67)

**Reply by falsealarmsnoozing** [reply]
Im getting the same error

**Reply by falsealarmsnoozing** [reply]
> I feel like im going crazy. I've followed your updated guide to the letter and keep running into this error here ![image](https://private-user-images.githubusercontent.com/55145070/384179945-30401e73-b179-4d7b-8ef8-5be5d61eeb67.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE3MjU4MDQsIm5iZiI6MTczMTcyNTUwNCwicGF0aCI6Ii81NTE0NTA3MC8zODQxNzk5NDUtMzA0MDFlNzMtYjE3OS00ZDdiLThlZjgtNWJlNWQ2MWVlYjY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE2VDAyNTE0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgwODM0MDE4OWQzYWQxMDQ2ZGMzZDU0ZWEzOTFhYjg1ZjMwZjQwMzlmNzlmZDVmYWI3MWUyNzM2OTFhYzFhOTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.wYP6uWVT2ALXfPOle2KxRb4KMt4ejd2tj89PdYJ-M9Q)

@SnowfallOCEC which version of youtube are you using? I tried the latest ones 19.44 and 19.55. Same error even with the instructions above. Could be that uYou is not yet updated to support this version may be????

**Reply by peppahacker** [reply]
Yeah, I think so, same problem.

**Comment by LucasEduard0** [comment]
Thank you very much for making this guide, I couldn't do it at all because I didn't understand anything about "decrypted IPA".

Thank you again!
